plutus-th/test/Spec.hs

```diff
@@ -7,30 +7,24 @@
 -- works.
 module Main (main) where
 
-import           Language.Plutus.TH
-
-import           Language.PlutusCore
-import qualified Language.PlutusCore.Pretty as PLC
+import           Common
 
 import           TestTH
 
-import           Language.Haskell.TH.Syntax
+import           Language.Plutus.TH
+
+import qualified Language.PlutusCore.Pretty as PLC
 
 import           Test.Tasty
-import           Test.Tasty.Golden
-
-import qualified Data.ByteString.Lazy as BSL
-import           Data.Text.Encoding   (encodeUtf8)
 
 main :: IO ()
-main = defaultMain tests
-
-golden :: String -> PlcCode -> TestTree
-golden name value = (goldenVsString name ("test/" ++ name ++ ".plc.golden") . pure . BSL.fromStrict . encodeUtf8 . PLC.docText . PLC.prettyPlcClassicDebug . getAst) value
+main = defaultMain $ runTestNestedIn ["test"] tests
 
+golden :: String -> PlcCode -> TestNested
+golden name value = (nestedGoldenVsDoc name . pure . PLC.prettyPlcClassicDebug . getAst) value
 
-tests :: TestTree
-tests = testGroup "Plutus TH frontend" [
+tests :: TestNested
+tests = testGroup "plutus-th" <$> sequence [
```

Comment:
I guess there is a specific reason why you do not use `testNested` here?

Reply:
I didn't want to introduce a pointless additional folder. We're already in `test`, I don't want another folder just to group all the top-level things.

Reply:
I'd very much prefer to keep golden files separately from machineries that use them. Even better: always keep golden files (nested or not) inside a "Golden" folder. Right now we sometimes interleave golden files with Haskell files and this looks like a mess and is not nice to navigate.

Reply:
They are separate - due to the nesting machinery I have `test/a`, `test/b` etc. I just don't see a point in having `test/golden/a`, `test/golden/b`.